src/System.Net.Http/src/System/Net/Http/Managed/AuthenticationHelper.Digest.cs

```diff
@@ -251,128 +244,143 @@ private static string ComputeHash(string data, string algorithm)
             }
         }
 
-        public class DigestResponse
+        internal class DigestResponse
         {
-            public readonly Dictionary<string, string> Parameters = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            internal readonly Dictionary<string, string> Parameters = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
 
             // Keep track of request values for this response.
-            public string Nonce;
-            public uint NonceCount;
+            internal uint NonceCount;
 
-            public DigestResponse(string challenge)
+            internal DigestResponse(string challenge)
             {
-                Nonce = null;
                 NonceCount = 1;
 
                 Parse(challenge);
             }
 
-            private unsafe string GetNextKey(ref char* p)
+            private string GetNextKey(ReadOnlySpan<char> p, out int parsedIndex)
```

Comment:
Is span needed at all?  If we're passing out the new index, why not just pass the original string around along with the next index?

Reply:
Yeah that should work as well, the fancy Span brainwashed me :P